modules/47_regipol/regiCarbonPrice/datainput.gms

```diff
@@ -22,32 +22,35 @@ $IFTHEN.emiMkt not "%cm_emiMktTarget%" == "off"
-
-*** Initializing European ETS historical and reference prices
-  loop(regi$regi_groupExt("EUR_regi",regi),
-    p47_taxemiMkt_init("2005",regi,"ETS")$(cm_startyear le 2005) = 0;
-    p47_taxemiMkt_init("2010",regi,"ETS")$(cm_startyear le 2010)  = 15*sm_DptCO2_2_TDpGtC;
-    p47_taxemiMkt_init("2015",regi,"ETS")$(cm_startyear le 2015)  = 8*sm_DptCO2_2_TDpGtC;
-***  p47_taxemiMkt_init("2020",regi,"ETS")$(cm_startyear le 2020)  = 41.28*sm_DptCO2_2_TDpGtC; !! 2018 =~ 16.5€/tCO2, 2019 =~ 25€/tCO2, 2020 =~ 25€/tCO2, 2021 =~ 53.65€/tCO2, 2022 =~ 80€/tCO2 -> average 2020 = 40€/tCO2 -> 40*1.032 $/tCO2 = 41.28 $/t CO2
-    p47_taxemiMkt_init("2020",regi,"ETS")$(cm_startyear le 2020)  = 30*sm_DptCO2_2_TDpGtC;
-
-*** Initializing European ESR historical and reference prices
-    p47_taxemiMkt_init("2020",regi,"ES")$(cm_startyear le 2020)  = 30*sm_DptCO2_2_TDpGtC;
-    p47_taxemiMkt_init("2020",regi,"other")$(cm_startyear le 2020)  = 30*sm_DptCO2_2_TDpGtC;
-  );
+*** initialize carbon taxes
+p47_taxemiMkt_init(t,regi,emiMkt) = 0;
+
+Execute_Loadpoint 'input_ref' p47_taxCO2eq_ref = pm_taxCO2eq;
+Execute_Loadpoint 'input_ref' p47_taxemiMkt_init = pm_taxemiMkt;
+
+*** copying taxCO2eq value to emiMkt tax parameter for years and regions that contain no pm_taxemiMkt value
+p47_taxemiMkt_init(t,regi,emiMkt)$(p47_taxCO2eq_ref(t,regi) and (NOT(p47_taxemiMkt_init(t,regi,emiMkt)))) = p47_taxCO2eq_ref(t,regi);
 
-*** intialize price trajectory after 2020 based on historical year prices for non policy scenarios 
-  if ( (cm_startyear le 2020),
-    p47_taxemiMkt_init(t,regi,emiMkt)$(t.val gt 2020)  = p47_taxemiMkt_init("2020",regi,emiMkt) + (cm_postTargetIncrease*sm_DptCO2_2_TDpGtC)*(t.val-2020);
+*** Overwrite historical prices for Europe if they are uncontrolled
+loop(regi$regi_groupExt("EUR_regi",regi),
+  if((cm_startyear le 2010),
+    p47_taxemiMkt_init("2010",regi,emiMkt) = 0;
+    p47_taxemiMkt_init("2010",regi,"ETS")  = 15*sm_DptCO2_2_TDpGtC;
+  );
+  if((cm_startyear le 2015),
+    p47_taxemiMkt_init("2015",regi,emiMkt) = 0;
+    p47_taxemiMkt_init("2015",regi,"ETS")  = 8*sm_DptCO2_2_TDpGtC;
+  );
+  if((cm_startyear le 2020),
+    p47_taxemiMkt_init("2020",regi,emiMkt) = 0;
+***  p47_taxemiMkt_init("2020",regi,"ETS")   = 41.28*sm_DptCO2_2_TDpGtC; !! 2018 =~ 16.5€/tCO2, 2019 =~ 25€/tCO2, 2020 =~ 25€/tCO2, 2021 =~ 53.65€/tCO2, 2022 =~ 80€/tCO2 -> average 2020 = 40€/tCO2 -> 40*1.032 $/tCO2 = 41.28 $/t CO2
+    p47_taxemiMkt_init("2020",regi,"ETS")  = 30*sm_DptCO2_2_TDpGtC;
+***    p47_taxemiMkt_init("2020",regi,"ES")   = 30*sm_DptCO2_2_TDpGtC;
+***    p47_taxemiMkt_init("2020",regi,"other")= 30*sm_DptCO2_2_TDpGtC;
   );
 
+*** intialize EUR price trajectory after 2020 based on historical year prices when no price is available
+  p47_taxemiMkt_init(t,regi,emiMkt)$(not(p47_taxemiMkt_init(t,regi,emiMkt)) and (t.val gt 2020)) = p47_taxemiMkt_init("2020",regi,emiMkt) + (1*sm_DptCO2_2_TDpGtC)*(t.val-2020);
```

Comment:
maybe add in the comment "by a yearly increase of 1$/tCO2"?

Reply:
I will add the comment. 
I just set it to 1$/tCO2 because it is kind of close to what we used in our NPI approximations for ECEMF. This is only a parameter initialization anyway, so it shouldn't affect much results, besides the fact that I am not starting from a completely flat or even zeros for future price values.

Reply:
https://github.com/remindmodel/remind/pull/1128/commits/e336def0a35e45a09b7181ea3e39a239e091a3e5